src/synapso/query_manager/query_manager.py

```
@@ -0,0 +1,29 @@
+class QueryManager:
+    def __init__(self, query_config: QueryConfig):
+        self.query_config = query_config
+        try:
+            self.vectorizer = VectorizerFactory.create_vectorizer(query_config.vectorizer_type)
+            self.reranker = RerankerFactory.create_reranker(query_config.reranker_type)
+            self.summarizer = SummarizerFactory.create_summarizer(query_config.summarizer_type)
+            self.meta_store = MetaStoreFactory.get_meta_store()
+            self.private_store = PrivateStoreFactory.get_private_store()
+            self.vector_store = VectorStoreFactory.get_vector_store()
+        except Exception as e:
+            raise RuntimeError(f"Failed to initialize QueryManager components: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)